### PR TITLE
Make gene filtering optional in core workflow

### LINKED ISSUE #303: Make gene filtering optional in core workflow
**Please provide some background on the proposed additions or changes.**

In the core module, we perform filtering of genes based on a minimum percent of cells genes must be detected in and a minimum mean gene expression. However, in the `processed.rds` files available on the portal, we do not do any filtering of the genes. 

**What are the changes that you are proposing?**

We should update the core module to match the filtering that is performed to generate the files on the portal by making gene filtering optional and setting the default to turn gene filtering off. 

**Please describe the proposed solution.**

To do this we should add a flag to control if gene filtering is true or false and set the default to false. Only then will the thresholds that are provided be used to perform filtering. 
This will also require an update in the documentation of the parameters. 

**What potential "gotchas" do we know of?**

**Additional context or questions**

### PR DESCRIPTION
**Pull request by cbethell**
**Issue Addressed**
Closes #303 

**What is the purpose of these changes?**

Per issue #303, this PR makes the gene filtering step in the `core-analysis/01-filter-sce` optional by adding a `filter_genes` option and logic to the script.

**What changes did you make?**

A `filter_genes` option was added to the `core-analysis/01-filter-sce.R` script, with a default of `FALSE`. Therefore, gene filtering will not be performed by default. If the `filter_genes` option _is_ provided, then the gene filtering threshold will be implemented for gene-level filtering.

The `README.md` file was also updated in this PR to include the added `filter_genes` parameter to the "filtering parameters" table.

**Any comments, concerns, or questions important for reviewers**
- Does this PR adequately address #303?

**Checklist**

Place an `x` in all boxes that you have completed.

- [x] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -251,6 +251,7 @@ Below are the parameters required to run either of the filtering methods.
 | `seed` | an integer to be used to set a seed for reproducibility when running the workflow | 2021 |
 | `filtering_method` | `filtering_method`, the specified filtering method which can be one of "miQC" or "manual". For more information on choosing a filtering method, see [Filtering low quality cells](./additional-docs/processing-information.md#filtering-low-quality-cells) in the [processing information documentation](./additional-docs/processing-information.md) | "miQC" |
 | `prob_compromised_cutoff` | the maximum probability of a cell being compromised as calculated by [miQC](https://bioconductor.org/packages/release/bioc/html/miQC.html), which is required when the `filtering_method` is set to `miQC` in the project metadata | 0.75 |
+| `filter_genes` | a binary value indicating whether or not to perform gene fitering | `FALSE` |
 | `gene_detected_row_cutoff` | the percent of cells a gene must be detected in; genes detected are filtered regardless of the `filtering_method` specified in the project metadata | 5 |
```
Comment (allyhawkins) [inline README.md:255]:
```suggestion
| `gene_detected_row_cutoff` | the percent of cells a gene must be detected in; genes detected are only filtered if `filter_genes` is set to `TRUE` | 5 |
```
We should modify this statement for this parameter and the `gene_means_cutoff` below.

File: core-analysis/01-filter-sce.R [hunk 2]
```diff
@@ -109,6 +110,12 @@ option_list <- list(
     help = "the selected filtered method -- can be miQC or manual; will be miQC
             by default"
   ),
+  optparse::make_option(
+    c("--filter_genes"),
+    action = "store_true",
+    default = FALSE,
+    help = "specifies whether or not to perform gene filtering"
```
Comment (allyhawkins) [inline core-analysis/01-filter-sce.R:117]:
```suggestion
    help = "specifies whether or not to perform gene filtering. Default is FALSE."
```

File: core-analysis/01-filter-sce.R [hunk 3]
```diff
@@ -237,16 +244,18 @@ if (opt$filtering_method == "manual") {
...
+  # Remove old gene-level rowData statistics and recalculate
+  drop_cols = colnames(rowData(filtered_sce)) %in% c('mean', 'detected')
+  rowData(filtered_sce) <- rowData(filtered_sce)[!drop_cols]
+  filtered_sce <- scater::addPerFeatureQC(filtered_sce)
+  
+  # Filter the genes (rows)
+  detected <-
+    rowData(filtered_sce)$detected > opt$gene_detected_row_cutoff
+  expressed <- rowData(filtered_sce)$mean > opt$gene_means_cutoff
+  filtered_sce <- filtered_sce[detected & expressed,]
```
Comment (allyhawkins) [inline core-analysis/01-filter-sce.R:257]: We should probably add a note in the metadata that genes are removed? And same for if genes are not removed?

**Review by allyhawkins — COMMENTED** [review]
This generally looks pretty good. I would add a note in the metadata about gene filtering and if it was performed or not. 

Also can you update the [overview section](https://github.com/AlexsLemonade/scpca-downstream-analyses/tree/development#core-analysis-overview)? 
In the `1.Filtering` section you should note that removal of genes is optional.

**Comment by cbethell** [comment]
Thanks for the review @allyhawkins -- I added a note to the metadata of the SCE object on whether or not genes have been filtered and updated the SCE content section of the `README`. I also made those additional doc updates you suggested, let me know what you think!

File: core-analysis/01-filter-sce.R [hunk 4]
```diff
@@ -237,16 +244,21 @@ if (opt$filtering_method == "manual") {
...
+  metadata(filtered_sce)$genes_filtered <- "genes filtered"
+} else {
+  metadata(filtered_sce)$genes_filterd <- "no genes filtered"
```
Comment (allyhawkins) [inline core-analysis/01-filter-sce.R:260]:
```suggestion
} 
metadata(filtered_sce)$genes_filtered <- opt$filter_genes
```
I think we can actually just do this and set it to true/false which will be the value of the argument so we can do it on one line.

File: core-analysis/01-filter-sce.R [hunk 5]
```diff
@@ -237,16 +244,21 @@ if (opt$filtering_method == "manual") {
...
+}
```
Comment (allyhawkins) [inline core-analysis/01-filter-sce.R:261]:
```suggestion
```
If you take my above suggestion.

**Review by allyhawkins — APPROVED** [review]
I just had one suggestion to simplify how we store the information in the metadata, but other than that looks good!

**Pull request merged**